spec/consensus/proposer-based-timestamp/pbts_001_draft.md

```diff
@@ -0,0 +1,268 @@
+`2f + 1` correct validators.
+   1. If there are `n` different timestamps in the  `precommit` messages, the proposer can use any subset of timestamps that add up to `2f + 1` 
+   of the voting power in order to compute the median.
+   1. If the validators decide in different rounds, the proposer can decide on which round the median computation is based. 
+1. **Liveness.** The liveness of the protocol:
+   1. does not depend on clock synchronization,
+   1. depends on bounded message delays.
+1. **Relation to real time.** There is no clock synchronizaton, which implies that there is **no relation** between the computed block `time` and real time.
+1. **Aggregate signatures.** As the `precommit` messages contain the local times, all these `precommit` messages typically differ in the time field, which **prevents** the use of aggregate signatures.
+
+## Suggested Proposer-Based Time
+
+### Outline
+
+An alternative approach to time has been discussed: Rather than having the validators send the time in the `precommit` messages, the proposer in the consensus algorithm sends its time in the `propose` message, and the validators locally check whether the time is OK (by comparing to their local clock). 
+
+This proposed solution adds the requirement of having synchronized clocks, and other implicit assumptions.
+
+### Comparison of the Suggested Method to the Old One
+
+1. **Fault tolerance.** Maintained in the suggested protocol. 
+2. **Effect of faulty validators.** Eliminated in the suggested protocol, 
+that is, the block `time` can be corrupted only in the extreme case when 
+`>2/3` of the validators are faulty.
+1. **Proposer influence on block time.** The proposer of the next block
+has less freedom when choosing the block time.
+   1. This scenario is eliminated in the suggested protocol, provided that there are `<1/3` faulty validators.
+   1. This scenario is still there.
+1. **Liveness.** The liveness of the suggested protocol:
+   1. depends on the introduced assumptions on synchronized clocks (see below),
+   1. still depends on the message delays (unavoidable).
+1. **Relation to real time.** We formalize clock synchronization, and obtain a **well-defined relation** between the block `time` and real time.
+1. **Aggregate signatures.** The `precommit` messages free of time, which **allows** for aggregate signatures.
+
+### Protocol Overview
+
+#### Proposed Time
+We assume that the field `proposal` in the `PROPOSE` message is a pair `(v, time)`, of the proposed consensus value `v` and the proposed time `time`. 
+
+#### Reception Step
+In the reception step at node `p` at local time `now_p`, upon receiving a message `m`:
+- **if** the message `m` is of type `PROPOSE` and satisfies `now_p - PRECISION <  m.time < now_p + PRECISION + MSGDELAY`, then mark the message as `timely`.  
```

Comment:
There has been some talk about giving the application some degree of control over the block times (https://github.com/tendermint/tendermint/issues/5911). Such control might be valuable when some transactions / loads require more time to process. In the case this idea was to come to any fruition would we redefine `timely` as:
```
now_p - PRECISION + BLOCKDELAY < m.time < now_p + PRECISION + MSGDELAY + BLOCKDELAY
```

Reply:
I don't think it interferes here. 

Rather it interferes with the `TimeoutPropose`: I guess the cleanest way to do this is that the proposer waits until the `BLOCKDELAY` has expired before sending the `PROPOSE` message.

In https://github.com/tendermint/spec/issues/265#issuecomment-803976562 I have estimated how `TimeoutPropose` needs to be computed to ensure monotonic block times, that is 
- `block(i+1).Time > block(i).Time`

This would need to be generalized to
- `block(i+1).Time > block(i).Time + BLOCKDELAY`

That is to compute `T2` in this case, we would need to add `BLOCKDELAY`. I guess we can already add a comment in the spec right now to be ready for such a change. What do you think?

Reply:
Thanks. Yeah this makes sense. If you want, you can mention it but I wouldn't say it's that the idea is well developed enough to warrant any changes to the math

Reply:
We have added a comment in the spec about how `BLOCKDELAY` can be incorporated in the future.